src/k8s-engine/lima.ts

```diff
@@ -449,28 +462,62 @@ export default class LimaBackend extends events.EventEmitter implements K8s.Kube
         await childProcess.spawnFile('tmutil', ['addexclusion', paths.lima]);
       }
     }
+    this.#externalInterfaceName = config.networks?.find(entry => (('lima' in entry) && ('interface' in entry)) )?.interface ?? INTERFACE_NAME;
+  }
+
+  protected async evalSymlinks(proposedPath: string) {
+    const dirs = proposedPath.split(path.sep);
+    let actualPath = '/';
+
+    for (let i = 1; i < dirs.length; i++) {
+      const currentPath = path.join(actualPath, dirs[i]);
+
+      try {
+        actualPath = path.resolve(actualPath, await fs.promises.readlink(currentPath));
+      } catch (_) {
+        // Possible failures:
+        // 1. currentPath not a symlink, just use it
+        // 2. currentPath doesn't exist, but presumably will in the future, so include it
+        // 3. Others: just use the currentPath as in (1.) and (2.) and ignore the cause of the failure.
+        actualPath = currentPath;
+      }
+    }
+
+    return actualPath;
+  }
+
+  protected checkMaxSocketLength(proposedPath: string) {
+    // See https://serverfault.com/questions/641347/check-if-a-path-exceeds-maximum-for-unix-domain-socket
+    // for an example of how to determine these values.
+    const socketLengthLimit = os.platform() === 'darwin' ? 103 : 107;
+
+    if (proposedPath.length > socketLengthLimit) {
+      console.log(`Specified path ${ proposedPath } symlink-expands to ${ proposedPath }`);
+      console.log(`The path ${ proposedPath } has ${ proposedPath.length } characters, over limit of ${ socketLengthLimit }`);
+      throw new Error(`Specified path ${ proposedPath } is too long, symlink-expands to ${ proposedPath }, ;exceeds limit by ${ proposedPath.length - socketLengthLimit } characters.`);
+    }
   }
 
-  protected updateConfigPortForwards(config: LimaConfiguration) {
+  protected async updateConfigPortForwards(config: LimaConfiguration) {
     let allPortForwards: Array<Record<string, any>> | undefined = config.portForwards;
 
     if (!allPortForwards) {
       // This shouldn't happen, but fix it anyway
       config.portForwards = allPortForwards = DEFAULT_CONFIG.portForwards ?? [];
     }
     const dockerPortForwards = allPortForwards?.find(entry => Object.keys(entry).length === 2 &&
-      entry.guestSocket === '/var/run/docker.sock');
+      entry.guestSocket === '/var/run/docker.sock' &&
+      ('hostSocket' in entry));
 
     if (!dockerPortForwards) {
+      const hostSocketPath = await this.evalSymlinks(`${ paths.lima }/${ MACHINE_NAME }/docker.sock`);
```

Comment:
I'm confused by this whole part of the code and don't really understand how it is supposed to work.

But it also doesn't seem to match what we talked about: that the `hostSocket` would be set to `docker`, and that RD would determinate the local socket name (with all symlinks resolved) using `limactl info`: https://github.com/rancher-sandbox/rancher-desktop/issues/910#issuecomment-963713946 (this is the only part of #910 that is relevant to 0.7.0).

A couple lines below it looks like this PR is actually reverting the desired implementation back to an older state. Did this get reverted during a rebase?